### PR TITLE
Hcc Update Entity Framework

### LINKED ISSUE #430: Update Entity Framework to Latest 6.xx
## Is your feature request related to a problem?  
Hotcakes Commerce is running on an older version of EF 6 and needs to be updated to the most current version.  

## Describe the solution you'd like  
Update EF to 6.4.4.  

## Describe alternatives you've considered  
None.  

## Additional context
N/A

### PR DESCRIPTION
**Pull request by ArielBlanco1990**
**Description of PR...**

The PR aims to migrate the version of the HCC EntityFramework from the current version 6.1.1 to version 6.4.4.

EF 6.4.0

The EF 6.4.0 runtime was released to NuGet in December 2019. The primary goal of EF 6.4 is to polish the features and scenarios that was delivered in EF 6.3. See list of important fixes on GitHub.

https://github.com/dotnet/ef6/milestone/14?closed=1

**Changes made**
___________________________________________________________________________________________________________________________________

- Update Hotcakes.Commerce.Dnn (.csproj file & packages.config)
- Update Hotcakes.Commerce.NoCms (.csproj file & packages.config)
- Update Hotcakes.Commerce (.csproj file & packages.config)
- Update Hotcakes.Web (.csproj file & packages.config)
- Update Hotcakes.Modules (.csproj file & packages.config)

Close #430

### CODE REVIEW

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:3]: Is this supposed to be here?  More important... Why is it here?  Any data access should be coming from the `Hotcakes.Commerce` project and the other libraries/projects should be calling into that.  Otherwise, there would have been a reference here that was replaced.  This one was added, so it wasn't necessary before and probably isn't necessary now.  

We need to be careful to keep the solution clean.  Do you have a good technical reason to add this circular dependency?  If not, it should be removed and re-tested.

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:274]: This has the same concern as my first comment.  However, if there is a good technical reason, then this will be approved.

File: Libraries/Hotcakes.Commerce.Dnn/packages.config
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.Dnn/packages.config:3]: What .NET version are the other projects running?  If we upgrade this one, we should upgrade them all, but only if it's required.  This update would overlap with and potentially break other features.  So, if the EF upgrade requires this, then we might need to update it everywhere now to avoid weird runtime issues later.

File: Libraries/Hotcakes.Commerce.NoCms/Hotcakes.Commerce.NoCms.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.NoCms/Hotcakes.Commerce.NoCms.csproj:3]: Same concern as the first comment in the other project.

File: Libraries/Hotcakes.Commerce.NoCms/Hotcakes.Commerce.NoCms.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.NoCms/Hotcakes.Commerce.NoCms.csproj:110]: Same concern as the first comment in the other project.

File: Libraries/Hotcakes.Commerce.NoCms/packages.config
Comment (WillStrohl) [inline Libraries/Hotcakes.Commerce.NoCms/packages.config:3]: Same concern as the first comment in the other project.

File: Libraries/Hotcakes.Web/Hotcakes.Web.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Web/Hotcakes.Web.csproj:3]: Same concern as the first comment in the other project.

File: Libraries/Hotcakes.Web/Hotcakes.Web.csproj
Comment (WillStrohl) [inline Libraries/Hotcakes.Web/Hotcakes.Web.csproj:179]: Same concern as the first comment in the other project.

File: Libraries/Hotcakes.Web/packages.config
Comment (WillStrohl) [inline Libraries/Hotcakes.Web/packages.config:4]: Same concern as the first comment in the other project.

File: Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj
Comment (WillStrohl) [inline Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj:3]: Same concern as the first comment in the other project.

File: Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj
Comment (WillStrohl) [inline Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj:4232]: What do these two lines of code do and/or what are they for?

File: Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj
Comment (WillStrohl) [inline Website/DesktopModules/Hotcakes/Hotcakes.Modules.csproj:4235]: Why does the `Hotcakes.Modules` project need to import the EF project when it didn't need to be imported before?

File: Website/DesktopModules/Hotcakes/packages.config
Comment (WillStrohl) [inline Website/DesktopModules/Hotcakes/packages.config:4]: Same concern as the first comment in the other project.

**Review by WillStrohl — CHANGES_REQUESTED** [review]
Please respond to my comments.  Everything might be fine as-is, but I need to be sure.

**Comment by ArielBlanco1990** [comment]
I'll go through all the comments and see if there's anything I can remove and get everything working correctly.  I understand the comments you have made.  I will check one by one.

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:3]: I have performed the EntityFramework update using the Nuget-Package-Manager tool, it seems that when performing the update automatically, unnecessary code segments have been added. I have made the updates again only changing what is necessary and everything has worked correctly.

Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:3]: Add a suggestion, <Ctrl+g>

New! Suggest specific code changes that the pull request author or assignees can immediately commit. You will be attributed in the commit.

Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:3]: I have performed the EntityFramework update using the Nuget-Package-Manager tool, it seems that when performing the update automatically, unnecessary code segments have been added. I have made the updates again only changing what is necessary and everything has worked correctly.

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:274]: I will make a new update with a cleaner solution than this. As I have mentioned, the changes were added automatically when using the Nuguet-Package-Manager tool.

Comment (WillStrohl) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:274]: @ArielBlanco1990 Okay. I figured that was the case, but let's make sure these updates are absolutely necessary.  Thank you!  :)

File: Libraries/Hotcakes.Commerce.Dnn/packages.config
Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/packages.config:3]: All the projects are using the same version of .Net. With the EntityFramework update in the solutions that have direct reference to this, it has already been automatically updated in all the solutions. I have checked inside each solution in the /bin/Debug/ folder and the EntityFramework dll that appears in all of them is the dll with version 6.4.4. I've done some tests and haven't noticed any problems apparently.
1 - I have generated a new solution and installed it in a clean DNN solution and everything has worked correctly.
2 - I have configured the store with sample products and everything is correct too.
3 - I have carried out payment tests and I have not noticed any problems either.

Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/packages.config:3]: This pull request needs some changes before being approved.

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:274]: This pull request needs some changes before being approved.

File: Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj
Comment (ArielBlanco1990) [reply Libraries/Hotcakes.Commerce.Dnn/Hotcakes.Commerce.Dnn.csproj:3]: This pull request needs some changes before being approved.

**Comment by ArielBlanco1990** [comment]
Please, do not approve this PR until I tell you it is ready for review. I have had some problems. After the new commits I have noticed something that should not happen.

**Comment by WillStrohl** [comment]
> Please, do not approve this PR until I tell you it is ready for review. I have had some problems. After the new commits I have noticed something that should not happen.

@ArielBlanco1990  I'm reviewing this PR now, per your internal note about completing the updates.  :)

**Review by WillStrohl — APPROVED** [review]
Approved.

**Comment by ArielBlanco1990** [comment]
Hi @WillStrohl : Yes after carrying out all the tests on my part it is ready to merge.

**Pull request merged**
